roles/airflow/tasks/main.yml

```diff
@@ -62,75 +62,107 @@
   become: yes
   become_user: airflow
   tags: airflow
 
-- name: ensure anacoda installation tmp dir exist on target system
-  file:
-    path: '/home/airflow/install/anaconda'
-    state: directory
-    owner: airflow
-  tags: airflow
-
-- name: ensure anacoda installation file is present on target system
+- name: ensure anacoda3 installer is present on target system
   copy:
     src: resources/Anaconda3-2018.12-Linux-x86_64.sh
-    dest: '/home/airflow/install/anaconda'
+    dest: /tmp/Anaconda3-2018.12-Linux-x86_64.sh
     owner: airflow
     mode: u+x,g+x
   tags: airflow
 
-- name: ensure anacoda is installed
-  shell: '/home/airflow/install/anaconda/Anaconda3-2018.12-Linux-x86_64.sh -b'
+- name: anaconda3 shell setup (clean-up)
+  blockinfile:
+    dest: /home/airflow/.bashrc
+    marker: "# {mark} ANSIBLE MANAGED BLOCK airflow-anaconda3"
+    block: |
+      . /home/airflow/anaconda3/etc/profile.d/conda.sh
+    state: absent
+  become: yes
+  become_user: airflow
+  tags: airflow
+
+- name: ensure anacoda3 is installed
+  shell: /tmp/Anaconda3-2018.12-Linux-x86_64.sh -b
   args:
     creates: /home/airflow/anaconda3
   become: yes
   become_user: airflow
-  register: ensure_anacoda_is_installed
+  register: anaconda3_installed
   tags: airflow
 
-- name: ensure conda is updated
+- name: ensure conda is up-to-date
   shell: '/home/airflow/anaconda3/bin/conda update -y conda'
   become: yes
   become_user: airflow
-  when: ensure_anacoda_is_installed.changed
+  when: anaconda3_installed is changed
   tags: airflow
 
 - name: ensure conda is initialized
   shell: '/home/airflow/anaconda3/bin/conda init bash'
   become: yes
   become_user: airflow
-  when: ensure_anacoda_is_installed.changed
+  when: anaconda3_installed is changed
   tags: airflow
 
-- name: ensure conda environment is created based on environment.yml file (first attempt)
-  shell: '/home/airflow/anaconda3/bin/conda env create --name o3 python=3.6 -f /home/airflow/o3/environment-linux.yml'
+- name: deactivate conda base env in in {{ bash_profile_file }}
+  blockinfile:
+    path: /home/airflow/{{ bash_profile_file }}
+    marker: "# {mark} ANSIBLE MANAGED BLOCK airflow-conda"
+    block: |
+      conda deactivate
   become: yes
   become_user: airflow
-  ignore_errors: true
-  register: ensure_conda_environment_is_created
-  when: ensure_anacoda_is_installed.changed
   tags: airflow
 
-- name: ensure conda environment is created based on environment.yml file
-  shell: '/home/airflow/anaconda3/bin/conda env create --name o3 python=3.6 -f /home/airflow/o3/environment-linux.yml'
+- name: create ~/.conda/pkgs
+  file:
+    path: /home/airflow/{{ item }}
+    state: directory
+  become: yes
+  become_user: airflow
+  with_items:
+    - .conda
+    - .conda/pkgs
+  register: dot_conda_dirs
+  tags: airflow
+
+- name: create ~/.conda/environments.txt and ~/anaconda3/envs/.conda_envs_dir_test
+  file:
+    path: /home/airflow/{{ item }}
+    state: touch
+    mode: 0664
```

Comment:
What about starting to use the much more readable way of doing this?

mode: "u=rw,g=rw,o=r"